com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ModuleDependencyChecker.java

```diff
@@ -0,0 +1,91 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.Module;
+
+public class ModuleDependencyChecker {
+
+    private Set<String> modulesIncludedInDeploy;
+    private Set<String> modulesExcludedFromDeploy;
+    private Set<String> modulesAlreadyDeployed;
+    CloudControllerClient client;
+
+    public ModuleDependencyChecker(CloudControllerClient client, DeploymentDescriptor descriptor, List<Module> includedModules,
+        List<Module> completedModules) {
+        this.client = client;
+
+        modulesIncludedInDeploy = includedModules.stream()
+            .map(Module::getName)
+            .collect(Collectors.toSet());
+        modulesAlreadyDeployed = completedModules.stream()
+            .map(Module::getName)
+            .collect(Collectors.toSet());
+        modulesExcludedFromDeploy = computeLeftOutModules(descriptor);
```

Comment:
Not a meaningful name. Try to rename it. The logic is complex here, so such misunderstanding names will make it more complex